src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java

```diff
@@ -710,32 +738,172 @@ public Kernel parseArgs(String... args) {
+            logger.atError().log("Unable to setup Nucleus launch parameters", e);
+        }
+    }
+
+    void initializeNucleusLifecycleConfig(String nucleusComponentName, ComponentRecipe componentRecipe) {
+        KernelConfigResolver kernelConfigResolver = context.get(KernelConfigResolver.class);
+        // Add Nucleus dependencies
+        Map<String, DependencyProperties> nucleusDependencies = componentRecipe.getDependencies();
+        if (nucleusDependencies == null) {
+            nucleusDependencies = Collections.emptyMap();
+        }
+        config.lookup(DEFAULT_VALUE_TIMESTAMP, SERVICES_NAMESPACE_TOPIC,
+                        nucleusComponentName, SERVICE_DEPENDENCIES_NAMESPACE_TOPIC)
+                .dflt(kernelConfigResolver.generateServiceDependencies(nucleusDependencies));
+
+        Topics nucleusLifecycle = config.lookupTopics(DEFAULT_VALUE_TIMESTAMP, SERVICES_NAMESPACE_TOPIC,
+                nucleusComponentName, SERVICE_LIFECYCLE_NAMESPACE_TOPIC);
+        if (!nucleusLifecycle.children.isEmpty()) {
+            logger.atDebug().log("Nucleus lifecycle has already been initialized");
+            return;
+        }
+        // Add Nucleus lifecycle (after config interpolation)
+        if (componentRecipe.getLifecycle() == null) {
+            return;
+        }
+        try {
+            Object interpolatedLifecycle = kernelConfigResolver.interpolate(componentRecipe.getLifecycle(),
+                    new ComponentIdentifier(nucleusComponentName, componentRecipe.getVersion()),
+                    nucleusDependencies.keySet(),
+                    config.lookupTopics(SERVICES_NAMESPACE_TOPIC).toPOJO());
+            nucleusLifecycle.replaceAndWait((Map<String, Object>) interpolatedLifecycle);
+            logger.atInfo().log("Nucleus lifecycle has been initialized successfully");
+        } catch (IOException e) {
+            logger.atError().log("Unable to initialize Nucleus lifecycle", e);
+        }
+    }
+
+    void initializeComponentStore(KernelAlternatives kernelAlts, String nucleusComponentName,
+                                  Semver componentVersion, Path recipePath,
+                                  Path unpackDir) throws IOException, PackageLoadingException {
+        // Copy recipe to component store
+        ComponentStore componentStore = context.get(ComponentStore.class);
+        ComponentIdentifier componentIdentifier = new ComponentIdentifier(nucleusComponentName, componentVersion);
+        Path destinationRecipePath = componentStore.resolveRecipePath(componentIdentifier);
+        if (!Files.exists(destinationRecipePath)) {
+            DeploymentService.copyRecipeFileToComponentStore(componentStore, recipePath, logger);
+        }
+
+        // Copy unpacked artifacts to component store
+        Path destinationArtifactPath = context.get(NucleusPaths.class).unarchiveArtifactPath(
+                componentIdentifier, DeviceConfiguration.DEFAULT_NUCLEUS_COMPONENT_NAME.toLowerCase());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

#### toLowerCase [](|2045d6a2-68fa-4cde-87a6-566ffddb2b76|)

The parameterless versions of `String.toLowerCase()` and `String.toUpperCase()` use the default locale of the JVM when transforming strings. This can have unintended results, including difficult-to-debug transient failures because case mapping differs based on locale. 

**Even if you aren’t writing code for a user experience that’s displayed to a customer**, system internal code often uses case folding to normalize strings for comparison.
Not doing so can missing locale lead to errors when running on systems that are configured to use a different system locale. This can include developer desktops or build servers (not just your fleet hosts): many hours have been spent debugging errors that only occur on a specific machine due to its locale configuration.

#### Recommended solutions:

Always pass in a `Locale`.

**Language or locale-dependent processing:**

Get the locale from the platform, use the language of the content, or configure it in your code and pass that to the method.
```
String firstName = &#34;Ichabod&#34;; 
Locale locale = // Where you obtain the locale from is platform dependent
String lowerCaseFirstName = firstName.toLowerCase(locale);
``` 

**Internal processing that isn’t locale-dependent:**

*Example: S3 bucket name*

Always specify the language/country-neutral locale `Locale.ROOT`. 

```
String imgTag = &#34;IMG&#34;; 
if (&#34;img&#34;.equals(imgTag.toLowerCase(Locale.ROOT)) {    
// do something 
}
```